docs/tutorials/query_casting.md

```diff
@@ -7,7 +7,7 @@ In older content this parameter is sometimes called `query` or `conditions`. For
 [require:Cast Tutorial.*get and set]
 ```
 
-When you execute the query using [`Query#exec()`](../api/query.html#query_Query-exec) or [`Query#then()`](../api/query.html#query_Query-then), Mongoose *casts* the filter to match your schema.
+When you execute the query using [`Query#exec()`](../api/query.html#query_Query-exec) or [`Query#then()`](../api/query.html#query_Query-then), Mongoose *casts* the filter to match your schema. Please, take note that casting will work *only* if the field is defined as mongoose.Schema.Types.ObjectId and not Types.ObjectId in the schema definition.
```

Comment:
Your claim is incorrect. Mongoose schemas do support both `mongoose.Schema.Types.ObjectId` and `mongoose.Types.ObjectId`.

```javascript
'use strict';

const mongoose = require('mongoose');

run().catch(err => console.log(err));

async function run() {
  await mongoose.connect('mongodb://127.0.0.1:27017/mongoose_test');

  const Test = mongoose.model('Test', new mongoose.Schema({
    myId: mongoose.Types.ObjectId
  }));
  const myId = new mongoose.Types.ObjectId();
  await Test.collection.insertOne({ myId });

  const doc = await Test.findOne({ myId: myId.toString() });
  console.log('Got', doc); // Prints the doc inserted with `insertOne()`
} 
```